src/pybind/mgr/telegraf/module.py

```diff
@@ -0,0 +1,314 @@
+import errno
+import json
+import socket
+import time
+from threading import Event
+
+from telegraf.basesocket import BaseSocket
+from telegraf.protocol import Line
+from mgr_module import MgrModule
+
+try:
+    from urllib.parse import urlparse
+except ImportError:
+    from urlparse import urlparse
+
+
+class Module(MgrModule):
+    COMMANDS = [
+        {
+            "cmd": "telegraf config-set name=key,type=CephString "
+                   "name=value,type=CephString",
+            "desc": "Set a configuration value",
+            "perm": "rw"
+        },
+        {
+            "cmd": "telegraf config-show",
+            "desc": "Show current configuration",
+            "perm": "r"
+        },
+        {
+            "cmd": "telegraf send",
+            "desc": "Force sending data to Telegraf",
+            "perm": "rw"
+        },
+        {
+            "cmd": "telegraf self-test",
+            "desc": "debug the module",
+            "perm": "rw"
+        },
+    ]
+
+    OPTIONS = [
+        {
+            'name': 'address',
+            'default': 'unixgram:///tmp/telegraf.sock',
+        },
+        {
+            'name': 'interval',
+            'default': 15
+        }
+    ]
+
+    fsid = None
```

Comment:
i still don't understand why we need to make `fsid` and `run` class variables.

Reply:
Yeah, this is kind of weird.